src/Features/CSharp/Portable/ConvertBetweenRegularAndVerbatimString/ConvertBetweenRegularAndVerbatimStringCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Immutable;
+using System.Composition;
+using System.Text;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.LanguageServices;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.VirtualChars;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.CSharp.ConvertBetweenRegularAndVerbatimString
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp), Shared]
+    internal class ConvertBetweenRegularAndVerbatimStringCodeRefactoringProvider
+        : AbstractConvertBetweenRegularAndVerbatimStringCodeRefactoringProvider<LiteralExpressionSyntax>
+    {
+        protected override bool IsInterpolation { get; } = false;
+
+        protected override bool IsAppropriateLiteralKind(LiteralExpressionSyntax literalExpression)
+            => literalExpression.Kind() == SyntaxKind.StringLiteralExpression;
+
+        protected override void AddSubStringTokens(LiteralExpressionSyntax literalExpression, ArrayBuilder<SyntaxToken> subStringTokens)
+            => subStringTokens.Add(literalExpression.Token);
+
+        protected override bool IsVerbatim(LiteralExpressionSyntax literalExpression)
+            => CSharpSyntaxFacts.Instance.IsVerbatimStringLiteral(literalExpression.Token);
+
+        protected override LiteralExpressionSyntax CreateVerbatimStringExpression(IVirtualCharService charService, StringBuilder sb, LiteralExpressionSyntax stringExpression)
+        {
+            sb.Append('@');
+            sb.Append(DoubleQuote);
```

Comment:
Nit: can this be implemented as:

```csharp
   protected override LiteralExpressionSyntax CreateVerbatimStringExpression(IVirtualCharService charService, StringBuilder sb, LiteralExpressionSyntax stringExpression)
   {
      sb.Append('@');
      return CreateRegularStringExpression(charService, sb, stringExpression);
   }
```

Reply:
not going to do this.  i think the current approach is clearer. thanks :)